model/structs.go

```diff
@@ -0,0 +1,60 @@
+package model
+
+type ImplementationStatus int32
+
+const (
+	Waiting ImplementationStatus = iota
+	Ready
+	Done
```

Comment:
Those are the three statuses that we're currently using but maybe it's a chance to add/change something? I'm thinking about something like:
**Proposed** //in PIP or whatever; still not final
**Waiting Implementation**
**Ready For Integration** //I'm skipping *Implementation* on purpose because it usually doesn't take long
**Integration Testing** //which can take a while and we have this exact case right now
**Released on Prod**
**Done** //and confirmed that we're getting this event type/field
**Decommissioned**

(yup, I know that without some nice GUI it will be hard to force anyone to use it but this level of information would tell the whole story about what's actually happening)

Reply:
And what be the flow for each of the objects that this status affects?

Reply:
I thought about it a little and came up with even more complex solution:

1. [optional] _**proposed**_ (someone from wants to have this field/event; there is no decision if it will be implemented)
2. **waiting for implementation** (we're planning to add this field/event)
3. **in development** (this  field/event is being worked on in this Sprint; it's also a fall back status if the change doesn't pass integration tests because of development issues)
4. **ready for integration** (development finished; waiting for integration tests)
5. [optional] **_integration testing_** (maybe it's an overkill, but it could be useful to show how long the testing can take)
6. **waiting for release** (integration tests passed)
7. **released** (our work is done)
8. **done** (event/field is used as intended; I know it's specific but it would also give us some pretty important information)
9. [optional] **_cancelled_** (field/event didn't make it to PROD)
10. [optional] **_decommissioned_** (field/event was on PROD but it shouldn't be used anymore)

The _standard flow_ would skip [optional] statuses and consist of 6 steps: 2, 3, 4, 6, 7, 8

Reply:
This should be viewed from 2 different perspectives:
1. Event implemented on AI site (`EventType` and `Field`)  _[Proposal -> Waiting ->Development -> Done-> Decommisioned; Cancelled]_
2. Event is being implemented on source system site (`EventTypeImplementation` and `FieldImplementation`) _[Waiting -> (opt)Development -> Integration Testing -> Done -> Decommisioned]_

Ad 1) A thinned version of what you proposed, without integration. Integration is source system only. For each source system.
ad 2) Source system has no proposal. The AI reference use is the proposal (`RecommendedUse` attributes). No actual need to know about development. Just integration and done are crucial.

@savoc What do you think?